testing-common/src/main/java/io/opentelemetry/instrumentation/testing/InstrumentationTestRunner.java

```diff
@@ -119,15 +121,29 @@ private <T extends Consumer<TraceAssert>> void waitAndAssertTraces(
     List<T> assertionsList = new ArrayList<>();
     assertions.forEach(assertionsList::add);
 
+    AtomicReference<AssertionError> assertionError = new AtomicReference<>();
     try {
       await()
-          .untilAsserted(() -> doAssertTraces(traceComparator, assertionsList, verifyScopeVersion));
+          .untilAsserted(
```

Comment:
Sorry for drive-by but as I was mentioned :) I don't remember details but do remember test failures being hard to debug before adding this hack with the bad traces, maybe there were some tests that had or have multiple assertions but again can't remember. I do remember being embarrassed to introduce awaitilitiy given the poor stack traces.

If the general test structure has changed or this corner case has been reduced enough it could still be fine, but want to reinforce the point that the stack traces can be important.

Reply:
Actually after these changes the behavior seems to be the same as with just using
`await().untilAsserted(() -> doAssertTraces(traceComparator, assertionsList, verifyScopeVersion));`
Junit5 seems to unwrap the `ConditionTimeoutException` and just shows the `AssertionFailedError` as the failure.
I think it is best not to make this change as it doesn't really resolve the underlying issue, which is to provide a stack trace that would let one easily determine which test failed. Since @anuraaga worked on this one thing that has changed is that we added `showStandardStreams = true` to `testLogging` to ease debugging some test failures. Because of that now when there is a test failure in CI it is easier to open the build scan rather than eyeball the test output and find the stack trace because there is a lot of output now. My guess would be that not having the full stack trace made it hard to quickly figure out which tests failed when you just looked at the test output.
To me the most annoying part of using awaitility `untilAsserted` here is that it really wants to make the tests pass. If there is a failure then it will try running the same asserts until the timeout. When there is a bug that makes a bunch of tests fail it takes a long time for the tests to complete. It might be better to change it so that it first waits for the requested amount of traces and then asserts on them once like the groovy tests do.